common/src/test/java/io/quarkiverse/tekton/common/utils/ParamsTest.java

```diff
@@ -25,14 +26,20 @@ void testCreateWithStringValue() {
 
     @Test
     void testCreateWithOddListValue() {
-        List<Param> param = Params.create(List.of("myKey=myValue"));
```

Comment:
In the original implementation of `Params` / `Params` test we had a single place where all the logic was implemented.
The pull request moves pieces of `Params` out of the class and thus we can no longer test splitting at `=` etc.

So, this test that covered splitting and mixing/matching now just tests key/values again and again.

Let's use the original implmentations and I'll raise an issue to better document it.

Reply:
I don't really see what you want to test. Can you elaborate please ? @iocanel

Reply:
I want to keep `ParamsTest` cases unaltered.

This pull request took things out of `Params` and thus the tests were changed so that they pass. I would like to change it back to how it was.

Reply:
I reverted the Params code and improved the PR. 
Fixed: b6236f3